lib/config/genconfig/hadeployconfig.go

```diff
@@ -0,0 +1,1102 @@
+}
+
+func (c *HaDeployConfigGen) PromptAutomate() (err error) {
+	err = c.PromptAutomateFqdn()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptAutomateFqdnRootCa()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptAutomateAdminPassword()
+	if err != nil {
+		return
+	}
+
+	c.DefaultAutomateConfigValues()
+
+	err = c.PromptAutomateNodes()
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (c *HaDeployConfigGen) PromptAutomateAdminPassword() (err error) {
+	adminPass, err := c.Prompt.InputPasswordRegex("Automate Dashboard Admin User Password", AUTOMATE_ADMIN_PASSWORD_REGEX)
+	if err != nil {
+		return
+	}
+	c.Config.InitAutomate().InitConfig().AdminPassword = adminPass
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefServer() (err error) {
+	err = c.PromptChefInfraServerFqdn()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptChefInfraServerFqdnRootCa()
+	if err != nil {
+		return
+	}
+
+	err = c.PromptChefInfraServerNodes()
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptOpenSearch() (err error) {
+	err = c.PromptOpenSearchNodes()
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptPostgresql() (err error) {
+	err = c.PromptPostgresqlNodes()
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefInfraServerNodes() (err error) {
+	chefServerNodeCount, err := c.Prompt.InputIntRange("No. of Chef Infra Server Nodes", 1, 100)
+	if err != nil {
+		return
+	}
+	c.Config.InitChefServer().InitConfig()
+	c.Config.ChefServer.Config.InstanceCount = fmt.Sprint(chefServerNodeCount)
+
+	hasCustomCerts, hasCustomCertsPerNode, err := c.PromptHaveCustomCerts(CHEF_INFRA_SERVER_NODETYPE)
+	if err != nil {
+		return err
+	}
+
+	c.Config.ChefServer.Config.EnableCustomCerts = hasCustomCerts
+
+	err = c.PromptChefInfraServerPriPubCerts(hasCustomCerts, hasCustomCertsPerNode)
+	if err != nil {
+		return err
+	}
+
+	c.Config.InitExistingInfra().InitConfig()
+	for i := 0; i < chefServerNodeCount; i++ {
+		err = c.PromptChefInfraServerNodeIpCerts(i, hasCustomCerts, hasCustomCertsPerNode)
+		if err != nil {
+			return err
+		}
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefInfraServerPriPubCerts(hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	if hasCustomCerts && !hasCustomCertsPerNode {
+		priKey, err := c.PromptCert("", CHEF_INFRA_SERVER_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+		c.Config.ChefServer.Config.PrivateKey = priKey
+
+		pubCert, err := c.PromptCert("", CHEF_INFRA_SERVER_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+		c.Config.ChefServer.Config.PublicKey = pubCert
+	}
+	if hasCustomCerts && hasCustomCertsPerNode {
+		c.Config.ChefServer.Config.InitCertsByIP()
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefInfraServerNodeIpCerts(i int, hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	ip, err := c.PromptNodeIp(fmt.Sprintf(PROMPT_NODE_IP, i+1))
```

Comment:
in case of more than 1 node count, the IPs should be unique, we can add checks for that

Reply:
I will consider this as feature enhancement for next time.